src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -177,24 +185,41 @@ public X509Certificate[] getAcceptedIssuers() {
                 }
             }
 
-            // initialize truststore
-            KeyStore ts = KeyStore.getInstance(KeyStore.getDefaultType());
+            KeyStore ts = KeyStore.getInstance(truststoreType);
             ts.load(null, null);
 
-            for (TruststoreConfig truststore : truststores) {
+            // initialize truststore with user provided path and pass
+            if (!truststorePath.isEmpty() && !truststorePass.isEmpty()) {
+                try (InputStream truststore = new FileInputStream(truststorePath.get())) {
+                    ts.load(truststore, truststorePass.get().get(passCharset).toCharArray());
```

Comment:
`.get().get(charset).toCharArray()` passes the bytes into a `new String()`, which would also be great to avoid doing if possible. Is there some other way to do the byte->char mapping using the charset without passing through a String constructor?